completor/create_output.py

```diff
@@ -4,449 +4,574 @@
+    Device and lateral branch numbers are changed accordingly if they exist.
+
+    Args:
+        lateral_number: The lateral number being worked on.
+        active_laterals: List of active lateral numbers.
+        df_device: Dataframe containing device data.
+        df_annulus: Dataframe containing annular data.
+
+    Returns:
+        Corrected device.
+    """
+    correction = max(active_laterals) - lateral_number
+    try:
+        df_device[Headers.BRANCH] += correction
+    except KeyError:
+        pass
+    try:
+        df_annulus[Headers.BRANCH] += correction
+    except KeyError:
+        pass
+    return df_device, df_annulus
+
+
+def _connect_lateral(well_name: str, lateral: Lateral, top: pd.DataFrame, weller_man: Well) -> pd.DataFrame:
+    """Connect lateral to main wellbore/branch.
+
+    The main branch can either have a tubing- or device-layer connected.
+    By default, the lateral will be connected to tubing-layer, but if connect_to_tubing is False,
+    it will be connected to device-layer.
+    Abort if it cannot find device layer at junction depth.
+
+    Args:
+        well_name: Well name.
+        lateral: Current lateral to connect.
+        top: DataFrame of first connection.
+        case: ReadCasefile object.
+        weller_man: Well object containing data from whole well.
+
+    Returns:
+        Tubing data with modified outsegment.
+
+    Raises:
+        CompletorError: If there is no device layer at junction of lateral.
+    """
+    if top.empty:
+        lateral.df_tubing.at[0, Headers.OUT] = 1  # Default out segment.
+        return lateral.df_tubing
+
+    first_lateral_in_top = top[Headers.TUBING_BRANCH].to_numpy()[0]
+    top_lateral = [lateral for lateral in weller_man.active_laterals if lateral.lateral_number == first_lateral_in_top][
+        0
+    ]
+    junction_measured_depth = top[Headers.TUBING_MEASURED_DEPTH].to_numpy()[0]
+    if junction_measured_depth > lateral.df_tubing[Headers.MEASURED_DEPTH][0]:
+        logger.warning(
+            "Found a junction above the start of the tubing layer, well %s, branch %s. "
+            "Check the depth of segments pointing at the main stem in schedulefile.",
+            well_name,
+            lateral.lateral_number,
+        )
+    if weller_man.case.connect_to_tubing(well_name, lateral.lateral_number):
+        layer_to_connect = top_lateral.df_tubing
+        measured_depths = top_lateral.df_tubing[Headers.MEASURED_DEPTH]
+    else:
+        layer_to_connect = top_lateral.df_device
+        measured_depths = top_lateral.df_device[Headers.MEASURED_DEPTH]
+    try:
+        if weller_man.case.connect_to_tubing(well_name, lateral.lateral_number):
+            # Since the junction_measured_depth has segment tops and layer_to_connect has grid block midpoints,
+            # a junction at the top of the well may not be found. Therefore, we try the following:
+            if (~(measured_depths <= junction_measured_depth)).all():
```

Comment:
Liter thinks this is a bool and does not have `.all()`, is type-hinting messed up or linter wrong?